plugins/woocommerce/src/Admin/Features/Settings/Transformer.php

```diff
@@ -0,0 +1,319 @@
+<?php
+/**
+ * WooCommerce Settings Data Transformer.
+ */
+
+declare( strict_types = 1 );
+
+namespace Automattic\WooCommerce\Admin\Features\Settings;
+
+/**
+ * Transforms WooCommerce settings data into a structured format with logical groupings.
+ */
+class Transformer {
+	/**
+	 * Current group being processed.
+	 *
+	 * @var array|null
+	 */
+	private ?array $current_group = null;
+
+	/**
+	 * Current checkbox group being processed.
+	 *
+	 * @var array|null
+	 */
+	private ?array $current_checkbox_group = null;
+
+	/**
+	 * Transform settings data.
+	 *
+	 * @param array $raw_settings Raw settings data.
+	 *
+	 * @return array Transformed settings data.
+	 */
+	public function transform( array $raw_settings ): array {
```

Comment:
Super duper optional but you can still keep this as static and this function can create its own class instance. I liked the `Transformer::transform( $pages );` we have previously 🥲